x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/HistogramGroupSource.java

```diff
@@ -54,6 +56,18 @@ public Type getType() {
         return Type.HISTOGRAM;
     }
 
+    @Override
+    public QueryBuilder getNextBucketsQuery(Object key) {
+        assert key instanceof Number;
+        return QueryBuilders.rangeQuery(field).gte(interval + ((Number)key).doubleValue());
+    }
+
+    @Override
+    public QueryBuilder getCurrentBucketQuery(Object key) {
+        assert key instanceof Number;
+        return QueryBuilders.rangeQuery(field).lt(((Number)key).doubleValue() - interval).gte(key);
```

Comment:
The reason for the `doubleValue()` is because the `key` in the histogram bucket is a `Double`. Though, I don't think it is ever anything not a whole number.